File: log.go
package unrealircd

// Log handles log operations
type Log struct {
	querier Querier
}

// Subscribe subscribes to log events. Any previous subscriptions are overwritten
func (l *Log) Subscribe(sources []string) (interface{}, error) {
	return l.querier.Query("log.subscribe", map[string]interface{}{
		"sources": sources,
	}, false)
}

// Unsubscribe unsubscribes from all log events
func (l *Log) Unsubscribe() (interface{}, error) {
	return l.querier.Query("log.unsubscribe", nil, false)
}

// GetAll gets past log events
func (l *Log) GetAll(sources []string) (interface{}, error) {
	params := map[string]interface{}{}
	if sources != nil {
		params["sources"] = sources
	}

	result, err := l.querier.Query("log.list", params, false)
	if err != nil {
		return nil, err
	}

	if res, ok := result.(map[string]interface{}); ok {
		if list, ok := res["list"]; ok {
			return list, nil
		}
	}

	return nil, nil
}

File: unrealircd-rpc-cs/UnrealIRCdRPC/Log.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using System.Text.Json;

namespace UnrealIRCdRPC
{
    // Log handles log operations
    public class Log
    {
        private readonly IQuerier _querier;

        public Log(IQuerier querier)
        {
            _querier = querier;
        }

        // Subscribe subscribes to log events
        public async Task<JsonElement?> SubscribeAsync(IEnumerable<string> sources)
        {
            return await _querier.QueryAsync("log.subscribe", new { sources }, false);
        }

        // Unsubscribe unsubscribes from log events
        public async Task<JsonElement?> UnsubscribeAsync()
        {
            return await _querier.QueryAsync("log.unsubscribe", null, false);
        }

        // GetAll gets log entries
        public async Task<JsonElement?> GetAllAsync(IEnumerable<string> sources)
        {
            return await _querier.QueryAsync("log.get", new { sources }, false);
        }
    }
}